pkg/postgres/repository.go

```diff
@@ -327,6 +329,87 @@ WHERE most_recent_event_moment < $1`,
 	return
 }
 
+func (a *RepositoryAgent) GetRecentlyActiveRuns(touchThreshold EasternTime) (runs map[martaapi.Line]map[martaapi.Direction][]Run, err error) {
+	rows, err := a.DB.Query(`
+SELECT runs.identifier, runs.run_group_identifier,
+  runs.corrected_line, runs.corrected_direction,
+  runs.most_recent_event_moment, runs.run_first_event_moment,
+  arrivals.identifier, arrivals.station, arrivals.arrival_time,
+  estimates.estimate_moment, estimates.estimated_arrival_time
+
+FROM runs
+JOIN arrivals
+  ON runs.identifier = arrivals.run_identifier
+JOIN estimates
+  ON arrivals.identifier = estimates.arrival_identifier
+
+WHERE runs.most_recent_event_moment > $1
+ORDER BY estimates.identifier ASC`,
+		touchThreshold,
+	)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to get active runs")
+		return
+	}
+
+	runsByIdentifier := map[string]Run{}
+	for rows.Next() {
+		var run Run
+		var arrival Arrival
+		var estimateMoment, estimatedArrivalTime EasternTime
+		err = rows.Scan(
+			&run.Identifier,
+			&run.RunGroupIdentifier,
+			&run.CorrectedLine,
+			&run.CorrectedDirection,
+			&run.MostRecentEventMoment,
+			&run.RunFirstEventMoment,
+			&arrival.Identifier,
+			&arrival.Station,
+			&arrival.ArrivalTime,
+			&estimateMoment,
+			&estimatedArrivalTime,
+		)
+		if err != nil {
+			err = errors.Wrapf(err, "failed to scan run")
+			return
+		}
+		run.setLineDirectionAndTrainID()
+
+		if seenRun, ok := runsByIdentifier[run.Identifier]; ok {
+			run = seenRun
```

Comment:
Since we reset the run here if we've seen it before, do we need to run line 377 run.setLineDirectionAndTrainID() before this? Not really an optimization or anything, just making sure I'm following the code right